internal/receiver/smartagentreceiver/converter/metrics.go

```diff
@@ -115,6 +112,6 @@ func fillNumberDatapoint(value sfx.Value, timestamp time.Time, dimensions map[st
 	attributes := dp.Attributes()
 	attributes.EnsureCapacity(len(dimensions))
 	for k, v := range dimensions {
-		attributes.Insert(k, pdata.NewAttributeValueString(v))
+		attributes.InsertString(k, v)
```

Comment:
More performant than the "Insert(k, new Attribute)" since we copy the attribute.